expression/builtin_cast_vec.go

```diff
@@ -118,3 +118,25 @@ func (b *builtinCastIntAsRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.
 func (b *builtinCastIntAsRealSig) vectorized() bool {
 	return true
 }
+
+func (b *builtinCastRealAsRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.Column) error {
+	if err := b.args[0].VecEvalReal(b.ctx, input, result); err != nil {
+		return err
+	}
+	n := input.NumRows()
+	f64s := result.Float64s()
+	conditionUnionAndUnsigned := b.inUnion && mysql.HasUnsignedFlag(b.tp.Flag)
```

Comment:
If `conditionUnionAndUnsigned` is false, perhaps we do not need to run the for loop below.

Reply:
yes, I'll update it